config/environments/development.rb

```diff
@@ -58,5 +58,7 @@
   # Allow disabling the webpack dev server from the settings
   config.webpack.dev_server.enabled = SETTINGS.fetch(:webpack_dev_server, true)
   config.webpack.dev_server.https = SETTINGS.fetch(:webpack_dev_server_https, false)
-  config.hosts << SETTINGS[:fqdn]
+
+  config.hosts += SETTINGS[:hosts]
+  config.hosts += (ENV['RAILS_DEVELOPMENT_HOSTS'] || '').split(',')
```

Comment:
I tested this, and it seems that the environment variable isn't automatically added by Rails to your `config.hosts` - it only worked for me if added explicitly like this.

Reply:
Turns out that's a Rails 7.0 feature: https://github.com/rails/rails/commit/8c50571808a8e6788eae493c6ffa739749b0ead0

Reply:
While you squash your PR, would you mind including a note that it's backporting a feature?
```suggestion
  # Backporting from Rails 7.0
  config.hosts += (ENV['RAILS_DEVELOPMENT_HOSTS'] || '').split(',')
```